tools/scoring/score.py

```diff
@@ -568,28 +606,124 @@ def project(normal,other):
     other_different = different.loc['other']
     return other_different
 
+# Do not consider solver variants for determining if a division is
+# competitive.
+# Uses solver_str(solver) to get the base version of solver.
+def is_division_competitive(solvers):
+    return len(set([solver_str(x) for x in solvers])) > 1
+
+
+# Biggest Lead Ranking.
+#
+# Computes the new global ranking based on the distance between the winner of a
+# division and the second solver in a division as defined in secion 7.3.1 of
+# the SMT-COMP'19 rules.
+#
+# Note: The function prints a list of sorted tuples starting with the first
+#       place (winner).
+#
+def biggest_lead_ranking(data):
+    start = time.time() if g_args.show_timestamps else None
+
+    data = group_and_rank_solver(data)
+    data = data[data.competitive == True]
+    scores = []
+    # TODO: use for (year, division) groupby(level ...)
+    for year, ydata in data.groupby('year'):
+        for division, div_data in ydata.groupby('division'):
+
+            # Skip non-competitive divisions
+            if not is_division_competitive(div_data.solver.unique()):
+                continue
+
+            assert len(div_data) >= 2
+            first = div_data.iloc[0]
+            second = div_data.iloc[1]
+            score = ((1 + first.correct) / (1 + second.correct))
+            scores.append(
+                (score, first.solver, second.solver, division))
+
+    scores_sorted = sorted(scores, reverse=True)
+    print('Biggest Lead Ranking (Score, 1st Solver, 2nd Solver, Division)')
+    for s in scores_sorted:
+        print(s)
+
+    if g_args.show_timestamps:
+        log('time biggest_lead_ranking: {}'.format(time.time() - start))
+
+
+# Largest Contribution Ranking.
+#
+# Compute the correctly solved score for the virtual best solver for a given
+# division if the results of 'solver' are excluded. This function corresponds
+# to function vbss(D,S) as defined in section 7.3.2 of the SMT-COMP'19 rules.
+#
+def vbss(division_data, solver):
+
+    # For VBS we only consider correctly solved benchmarks
+    data = division_data[(division_data.solver != solver)
+                         & (division_data.correct > 0)
+                         & (division_data.error == 0)]
+
+    sort_columns = ['benchmark', 'score_correct', 'wallclock_time']
```

Comment:
In the rules document it's cpu time, perhaps something to discuss.

Reply:
Ah right, fixed it.